internal/util/auxutil.go

```diff
@@ -0,0 +1,88 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EtcdClusterIsReady returns true is condition "Ready" has status equals to "True", otherwise false.
+func EtcdClusterIsReady(cluster *etcdaenixiov1alpha1.EtcdCluster) bool {
+	idx := slices.IndexFunc(cluster.Status.Conditions, func(condition metav1.Condition) bool {
+		return condition.Type == etcdaenixiov1alpha1.EtcdConditionReady
+	})
+	if idx == -1 {
+		return false
+	}
+	return cluster.Status.Conditions[idx].Status == metav1.ConditionTrue
+}
+
+// FillConditions fills EtcdCluster .status.Conditions list with all available conditions in "False" status.
+func FillConditions(cluster *etcdaenixiov1alpha1.EtcdCluster) {
+	conditions := []string{
+		etcdaenixiov1alpha1.EtcdConditionInitialized,
+		etcdaenixiov1alpha1.EtcdConditionReady,
+	}
+
+	for _, c := range conditions {
+		SetCondition(cluster, c, false)
+	}
+}
+
+// SetCondition sets the condition of defined type to desired state.
+func SetCondition(
+	cluster *etcdaenixiov1alpha1.EtcdCluster,
+	conditionType string,
+	conditionStatus bool,
+) {
+	status := metav1.ConditionFalse
+	if conditionStatus {
+		status = metav1.ConditionTrue
+	}
+
+	idx := slices.IndexFunc(cluster.Status.Conditions, func(condition metav1.Condition) bool {
+		return condition.Type == conditionType
+	})
+
+	reason, message := conditionReasonMessage(conditionType, status)
+	condition := metav1.Condition{
+		Type:               conditionType,
+		Status:             status,
+		ObservedGeneration: cluster.Generation,
+		LastTransitionTime: metav1.Now(),
+		Reason:             reason,
+		Message:            message,
+	}
+
+	if idx == -1 {
+		cluster.Status.Conditions = append(cluster.Status.Conditions, condition)
+		return
+	}
+	if cluster.Status.Conditions[idx].Status == status {
+		return
+	}
+	cluster.Status.Conditions[idx] = condition
+}
+
+func conditionReasonMessage(conditionType string, status metav1.ConditionStatus) (string, string) {
+	var (
+		reason, message string
+	)
+
+	switch {
+	case conditionType == etcdaenixiov1alpha1.EtcdConditionInitialized && status == metav1.ConditionFalse:
+		reason = string(etcdaenixiov1alpha1.EtcdCondTypeInitStarted)
+		message = string(etcdaenixiov1alpha1.EtcdInitCondNegMessage)
+	case conditionType == etcdaenixiov1alpha1.EtcdConditionInitialized && status == metav1.ConditionTrue:
+		reason = string(etcdaenixiov1alpha1.EtcdCondTypeInitComplete)
+		message = string(etcdaenixiov1alpha1.EtcdInitCondPosMessage)
+	case conditionType == etcdaenixiov1alpha1.EtcdConditionReady && status == metav1.ConditionFalse:
+		reason = string(etcdaenixiov1alpha1.EtcdCondTypeStatefulSetNotReady)
+		message = string(etcdaenixiov1alpha1.EtcdReadyCondNegMessage)
+	case conditionType == etcdaenixiov1alpha1.EtcdConditionReady && status == metav1.ConditionTrue:
+		reason = string(etcdaenixiov1alpha1.EtcdCondTypeStatefulSetReady)
+		message = string(etcdaenixiov1alpha1.EtcdReadyCondPosMessage)
+	}
```

Comment:
This won't let add more options for negative statuses. For example after writing custom pod controller, reconciler can fail on creating pods because of resource quota or range limit. I think we will need to add custom message here. Let's pass everywhere message as just string without consts

Reply:
It make sense, however at the moment it will be over engineering to take into account such cases, which are not even designed.
From my perspective, in the future it will be reasonable to make setting of condition as something like wrapper on the cluster's "phase" processing functions. This will allow us to make conditions' reasons & messages accurate and meaningful.
Also, I'd prefer not to put all possible failure reason-message into "Ready" state, but add conditions like "PodsScheduled" etc.

Reply:
It's not over engineering. It will simplify code:
- there won't be implicit selecting of messages for type
- it will be simpler to add new message for any type at any time

Reply:
Hm. Ok then, wdyt about adding condition builder? Which will result in something like

```go
// etcdcluster_reconciler.go
...

util.setCondition(cluster, condition().
    WithType(...).
    WithStatus(...).
    WithReason(...).
    WithMessage(...)

...
```

Reply:
I like your suggestion :) 
It will be simpler as of helpers logic, it's just one parameter and it will be as powerful as 4 separate parameters